apps/wallet/src/ui/app/pages/dapp-tx-approval/TransactionTypeCard.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import LoadingIndicator from '_components/loading/LoadingIndicator';
+import { useFormatCoin } from '_hooks';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+
+import st from './DappTxApprovalPage.module.scss';
+
+type TransactionTypeCardProps = {
+    label: string;
+    content: string | number | null;
+    loading: boolean;
+};
+
+const GAS_ESTIMATE_LABEL = 'Estimated Gas Fees';
+
+export function TransactionTypeCard({
+    label,
+    content,
+    loading,
+}: TransactionTypeCardProps) {
+    const isGasEstimate = label === GAS_ESTIMATE_LABEL;
```

Comment:
We generally don't want to use the label for behavior like this, it won't work well for (eventual) i18n.